rlp/encode.go

```diff
@@ -206,6 +211,22 @@ func writeBigIntNoPtr(val reflect.Value, w *encBuffer) error {
 	return nil
 }
 
+func writeU256IntPtr(val reflect.Value, w *encBuffer) error {
+	ptr := val.Interface().(*uint256.Int)
+	if ptr == nil {
+		w.str = append(w.str, 0x80)
+		return nil
+	}
+	ptr.EncodeRLP(w)
```

Comment:
This is weird. We shouldn't rely on the method in uint256 here. Instead, we can just put the encoding logic in this package.

Reply:
Ok, fixed, please check